src/provider/Provider.ts

```diff
@@ -599,38 +596,26 @@ export class Provider {
       ? this.getEndpointURL(serviceEndpoints, 'computeResult').urlPath
       : null
 
-    const nonce = await this.getNonce(
-      providerUri,
-      accountId,
-      signal,
-      providerEndpoints,
-      serviceEndpoints
-    )
-
-    let signatureMessage = accountId
+    const nonce = Date.now()
+    let signatureMessage = consumerAddress
     signatureMessage += jobId
     signatureMessage += index.toString()
     signatureMessage += nonce
-    const signature = await this.signProviderRequest(web3, accountId, signatureMessage)
+    const signature = await this.signProviderRequest(
+      web3,
+      consumerAddress,
+      signatureMessage
+    )
 
-    let consumeUrl = computeResultUrl
-    consumeUrl += `?consumerAddress=${accountId}`
-    consumeUrl += `&jobId=${jobId}`
-    consumeUrl += `&index=${String(index)}`
-    consumeUrl += (signature && `&signature=${signature}`) || ''
+    let resultUrl = computeResultUrl
+    resultUrl += `?consumerAddress=${consumerAddress}`
+    resultUrl += `&jobId=${jobId}`
+    resultUrl += `&index=${index.toString()}`
+    resultUrl += `&nonce=${nonce}`
+    resultUrl += (signature && `&signature=${signature}`) || ''
 
-    if (!computeResultUrl) return null
-    try {
-      if (document) {
-        await downloadFileBrowser(consumeUrl)
-      } else {
-        return await downloadFile(consumeUrl, index)
-      }
-    } catch (e) {
-      LoggerInstance.error('Error getting job result')
-      LoggerInstance.error(e)
-      throw e
-    }
+    if (!resultUrl) return null
+    return resultUrl
```

Comment:
This would do the same thing as those 2 lines and multiple return statements:

```ts
    return resultUrl || null
```

But looks like `resultUrl` will always be defined as we are adding strings to it above so would end up with some string like `null?consumerAddress=&jobId=`. So not sure what the check for empty `resultUrl` would give us here